### DISCUSSION TITLE
Análise da quantificação de recuo e avanço de geleiras

### DISCUSSION THREAD

**Discussion by AlissonSteffens** [lia-univali/cassie#72] [Ideas]
Esta thread é volta a discussões sobre como implementar esta análise no CASSIE.
[repositório GEE](https://code.earthengine.google.com/?accept_repo=users/229149392119gm/glacierMapping)

**Comment by gandrat** [comment]
Ok, estou por aqui.

**Comment by vinigmartins** [comment]
Link para o repositório no GEE em que estou trabalhando [glacierMapping](https://code.earthengine.google.com/?accept_repo=users/229149392119gm/glacierMapping)

Adicionei uma pasta chamada "Modules" com scripts que podem ser reutilizáveis em outros scripts, como por exemplo calculo de diferença normaliza e obtenção do histograma da imagem.

Assim que obter bons resultados em testes, informarei por aqui.

**Comment by vinigmartins** [comment]
Realizei testes com a operação DOS (Dark Object Subtraction).
Como suspeitava, ao realizar a subtração do pixel mais escuro quando este é negativo ele passa a somar, tornando a operação menos eficaz.

![image](https://user-images.githubusercontent.com/30875400/115163339-fadc5800-a07e-11eb-9932-891d48af88e9.png)

Atualmente estou procurando métodos para contornar isso e formas para automatizar testes em varias regiões.

**Reply by vinigmartins** [reply]
Realizei novamente testes com a operação DOS, embora existem valores negativos são muito próximos de zero, não impactando na correção feita pela operação.

Imagem Original | Imagem DOS
-------------------|-----------------
![image](https://user-images.githubusercontent.com/30875400/120366126-a8c95c00-c2e5-11eb-8793-5eb7edf423eb.png) | ![image](https://user-images.githubusercontent.com/30875400/120366176-bbdc2c00-c2e5-11eb-939b-9bb296d1cf33.png)

NDWI Original | NDWI DOS
-----------------|-----------------
![image](https://user-images.githubusercontent.com/30875400/120368945-090dcd00-c2e9-11eb-8bd7-52baa84d3322.png) | ![image](https://user-images.githubusercontent.com/30875400/120368835-eda2c200-c2e8-11eb-84b5-c5b8c560c62f.png)

**Reply by AlissonSteffens** [reply]
Aparentemente o impacto das sombras é menor. Certamente vale a pena manter a operação

**Comment by vinigmartins** [comment]
## Progresso com técnica de segmentação [Tests/Segmentation](https://code.earthengine.google.com/?accept_repo=users/229149392119gm/glacierMapping)

Para a segmentação da geleira foi utilizado o método Weka KMeans do GEE.

Sequência do teste realizado:
1. Obtenção do NDWI e NDSI.
2. Otsuthreshold para limiarização de ambos.
3. Limiarização do NDWI e NDSI.
4. Treino de amostra para o cluster utilizando a imagem clipada com a mascara gerada pela operação lógica (ndwi limiarizado AND ndsi limiarizado).
5. Cluster de duas classes com o treino gerado anteriormente.
6. Resultado gerado pelo fechamento(operação morfológica) do cluster seguido pela operação lógica (fechamento AND mascara do treino)

### Geleira divisa com água 
Image RGB | Resultado
--------------|------------
![geleiraComAgua_rgb](https://user-images.githubusercontent.com/30875400/124494221-f32e7480-dd8c-11eb-8a65-33db1fa17fcd.png) | ![geleiraComAgua_result](https://user-images.githubusercontent.com/30875400/124494279-06414480-dd8d-11eb-90e3-a989655d59ec.png)


### Geleira divisa com gelo
Image RGB | Resultado
-----------------|-----------------
![geleiraComGelo_rgb](https://user-images.githubusercontent.com/30875400/124494596-6e902600-dd8d-11eb-8ea6-ee7c2b11a61c.png) | ![geleiraComGelo_result](https://user-images.githubusercontent.com/30875400/124494624-76e86100-dd8d-11eb-8e70-daa944b87e28.png)

### Conclusão
Independentemente da(s) banda(s) para amostra estando com ou sem máscara para o treino do cluster, em regiões onde não possui água a segmentação não é suficiente para a identificação de geleiras.
